### PR TITLE
Update SST packages to 13.1.0

### PR DESCRIPTION
**Pull request by berquist**

### CODE REVIEW

File: var/spack/repos/builtin/packages/sst-core/package.py [hunk 1]
```diff
@@ -61,7 +62,7 @@ class SstCore(AutotoolsPackage):
     variant("preview", default=False, description="Preview build with deprecated features removed")
     variant("profile", default=False, description="Enable performance profiling of core features")
 
-    depends_on("python", type=("build", "run", "link"))
+    depends_on("python@:3.12", type=("build", "run", "link"))
```
Comment (bernhardkaindl) [inline var/spack/repos/builtin/packages/sst-core/package.py]:
```suggestion
    depends_on("python@:3.11", type=("build", "run", "link"))
```
@berquist You comitted this with: "SST does not work with Python 3.12 yet"

If I'm not totally mistaken, you'd have to update this commit to use `python@:3.11` instead.

File: var/spack/repos/builtin/packages/sst-elements/package.py [hunk 2]
```diff
@@ -56,7 +57,7 @@ class SstElements(AutotoolsPackage):
     variant("otf", default=False, description="Build with OTF")
     variant("otf2", default=False, description="Build with OTF2")
 
-    depends_on("python", type=("build", "run"))
+    depends_on("python@:3.12", type=("build", "run"))
```
Comment (bernhardkaindl) [inline var/spack/repos/builtin/packages/sst-elements/package.py]:
```suggestion
    depends_on("python@:3.11", type=("build", "run"))
```
@berquist You committed this with: "SST does not work with Python 3.12 yet"

If I'm not totally mistaken, you'd have to update this commit to use `python@:3.11` instead.

**Review by bernhardkaindl — COMMENTED** [review]
I think I found two places where I think you meant `python@:3.11` instead of `:12` as you wrote that SST doesn't work with Python 3.13 yet.

@naromero77: Could you review this PR within the next 7 days?

**Comment by berquist** [comment]
Sorry, brain stuck in Python-land.  Now it fails properly.
```
$ spack concretize --force
==> Error: concretization failed for the following reasons:

   1. Cannot select a single "version" for package "python"
   2. Cannot satisfy 'python@:3.11'
   3. Cannot satisfy 'python@3:3.11'
   4. Cannot satisfy 'python@3.9:3.11'
   5. Cannot satisfy 'python@3.8:3.11'
   6. Cannot satisfy 'python@3.12:'
   7. Cannot satisfy 'python@3.12'
   8. Cannot satisfy 'python@:3.11'
        required because sst-core depends on python@:3.11
          required because sst-core@13.1.0 ^python@3.12 requested from CLI
   9. Cannot satisfy 'python@3.12'
        required because sst-core@13.1.0 ^python@3.12 requested from CLI
   10. Cannot satisfy 'python@:3.11' and 'python@3.12
        required because sst-core@13.1.0 ^python@3.12 requested from CLI
        required because sst-core depends on python@:3.11
          required because sst-core@13.1.0 ^python@3.12 requested from CLI
   11. Cannot satisfy 'python@3.12' and 'python@:3.11
        required because sst-core@13.1.0 ^python@3.12 requested from CLI
        required because sst-core depends on python@:3.11
          required because sst-core@13.1.0 ^python@3.12 requested from CLI
```

File: var/spack/repos/builtin/packages/sst-core/package.py [hunk 3]
```diff
@@ -73,6 +74,12 @@ class SstCore(AutotoolsPackage):
     depends_on("m4", type="build", when="@master:")
     depends_on("gettext")
 
+    sanity_check_is_file = [
+        join_path("bin", "sst-config"),
+        join_path("bin", "sst-test-core"),
+        join_path("bin", "sst-test-elements"),
+    ]
```
Comment (naromero77) [inline var/spack/repos/builtin/packages/sst-core/package.py]: Could you please add a comment as to why this is necessary? I am assuming this is due to some shortcoming in SST's autotools set-up?

Comment (berquist) [reply var/spack/repos/builtin/packages/sst-core/package.py]: It's not really necessary so I've removed it.

**Review by naromero77 — APPROVED** [review]
LGTM

**Pull request merged**